presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -329,6 +333,14 @@ public static PlanNode pushPredicateIntoTableScan(
             return new ValuesNode(idAllocator.getNextId(), node.getOutputVariables(), ImmutableList.of());
         }
 
+        RowExpression optimizedRemainingExpression = new RowExpressionOptimizer(metadata).optimize(translatedExpression.getRemainingExpression(), OPTIMIZED, session.toConnectorSession());
```

Comment:
@bhhari Looks like there is some redundancy here. The expression is being split into tuple domain and remaining predicate and these parts are being checked for "always-false" separately. Let's just optimize the whole expression and check the result.

Looks like `translatedExpression.getRemainingExpression(),` can be replaced by `expression` and logic using `translatedExpression` can be removed. 

```
        // optimize rowExpression and return ValuesNode if false. e.g. 0=1 => false, 1>2 => false, (a = 1 and a = 2) => false
        RowExpression optimizedRemainingExpression = new RowExpressionOptimizer(metadata).optimize(expression, OPTIMIZED, session.toConnectorSession());
        if (optimizedRemainingExpression instanceof ConstantExpression) {
            ConstantExpression constantExpression = (ConstantExpression) optimizedRemainingExpression;
            if (FALSE_CONSTANT.equals(constantExpression) || constantExpression.getValue() == null) {
                return new ValuesNode(idAllocator.getNextId(), node.getOutputVariables(), ImmutableList.of());
            }
        }
```

There is still a question about the case when optimized expression is not always-false, but rather uses fewer columns when original expression. This will generate an error, right?

Reply:
yes mentioned that in the commit message, it looks like if we use the optmization level as "Optimize" we can not serialize it back. This commit only solves the case where its null/false

Reply:
@bhhari  Are you planning to add another commit to fix the remaining issue. Something similar to the original fix could work. E.g. prune the predicate columns on the worker after optimizing the remaining expression.

Reply:
I m going to update the same commit, prune it during the planning